### DISCUSSION TITLE
survey text data.

### DISCUSSION THREAD

**Discussion by rbruno1** [jspsych/jsPsych#1987] [Q&A] [answered]
Hi, I am trying to add a "correct_response" function to the data section of my trials. Here is what I've got:
```js
const POD_1_response = {
  type: 'survey-text',
  questions: [
    {prompt: "What word was she about to say? (no caps!)"}
  ],
  post_trial_gap: 500,
  data: {condition: 'Bucket', correct_response: "bucket"},
  on_finish: function(data) {
    data.correct = jsPsych.pluginAPI.compareKeys(data.response, data.correct_response);
  }
}
```
and here is what the data display shows:
```js
  "rt": 4356,
  "response": {
    "Q0": "bucket"
  },
  "condition": "Bucket",
  "correct_response": "bucket",
  "trial_type": "survey-text",
  "trial_index": 5,
  "time_elapsed": 16094,
  "internal_node_id": "0.0-4.0-1.0"
```

How do i got the correct_response function to recognize the survey-text responses?

**Comment by becky-gilbert** [comment, marked as answer]
Hi @rbruno1, there are two issues here. First, the `compareKeys` function is just meant for checking whether two _keyboard key presses_ are the same, so I wouldn't use that function to compare two text responses. Second, to access the participant's response in the `on_finish` function, you'll need to use `data.response.Q0`, because `data.response` is an object that contains key/value pairs for each question, and you're trying to access the response to the first (only) question, which is automatically called "Q0".  

So try this in your trial's `on_finish` function:
```js
data.correct = data.response.Q0 == data.correct_response;
```
And here's another tip: JavaScript has the [`toLowerCase` function](https://www.w3schools.com/jsref/jsref_tolowercase.asp) for converting strings to lowercase, so you could use that just in case someone isn't following your instructions, or doesn't notice that they've turned caps lock on. 
```js
data.correct = data.response.Q0.toLowerCase() == data.correct_response;
```
And there are some other fancy things you could do with JavaScript, like remove spaces using [`trim`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim) and [`replace` to remove non-letter characters](https://www.codegrepper.com/code-examples/javascript/remove+non+letters+from+string+javascript) 😃 

I hope this helps.
